src/geode/inspector/criterion/degeneration/solid_degeneration.cpp

```diff
@@ -41,6 +44,83 @@ namespace geode
                   verbose }
         {
         }
+
+        bool is_mesh_degenerated() const final
+        {
+            if( this->detail::DegenerationImpl<
+                    SolidMesh< dimension > >::is_mesh_degenerated() )
+            {
+                return true;
+            }
+            for( const auto polyhedron_id :
+                Range{ this->mesh().nb_polyhedra() } )
+            {
+                if( polyhedron_is_degenerated( polyhedron_id ) )
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        index_t nb_degenerated_polyhedra() const
+        {
+            index_t counter{ 0 };
+            for( const auto polyhedron_id :
+                Range{ this->mesh().nb_polyhedra() } )
+            {
+                if( polyhedron_is_degenerated( polyhedron_id ) )
+                {
+                    counter++;
+                }
+            }
+            return counter;
+        }
+
+        std::vector< index_t > degenerated_polyhedra() const
+        {
+            std::vector< index_t > wrong_polyhedra;
+            for( const auto polyhedron_id :
+                Range{ this->mesh().nb_polyhedra() } )
+            {
+                if( polyhedron_is_degenerated( polyhedron_id ) )
+                {
+                    wrong_polyhedra.push_back( polyhedron_id );
+                }
+            }
+            return wrong_polyhedra;
+        }
+
+    private:
+        bool polyhedron_is_degenerated( index_t polyhedron_id ) const
+        {
+            const auto& mesh = this->mesh();
+            const auto& p1 =
+                mesh.point( mesh.polyhedron_vertex( { polyhedron_id, 0 } ) );
+            for( const auto i :
+                LRange{ 1, mesh.nb_polyhedron_vertices( polyhedron_id ) - 2 } )
+            {
+                const auto& p2 = mesh.point(
+                    mesh.polyhedron_vertex( { polyhedron_id, i } ) );
+                const auto& p3 =
+                    mesh.point( mesh.polyhedron_vertex( { polyhedron_id,
+                        static_cast< local_index_t >( i + 1 ) } ) );
+                const auto& p4 =
+                    mesh.point( mesh.polyhedron_vertex( { polyhedron_id,
+                        static_cast< local_index_t >( i + 2 ) } ) );
+                const Tetrahedron polyhedron_tetra{ p1, p2, p3, p4 };
+                if( tetrahedron_volume( polyhedron_tetra ) > global_epsilon )
+                {
+                    return false;
```

Comment:
Maybe I will check the sum of signed volumes of decomposing tetrahedra (i.e. ~ the volume of the polyhedron).